Release 5.0.0 (#16)
smartdevicelink/sdl_core_documentation: docs/Software Architecture Document/Components View/index.md

The view is represented by module and subsystem diagrams that show the system's export and import relationships. The Components View diagram and its elements description please see below.

> *Note*: UML notation for this Components View diagram is extended: both component and it's interfaces are highlighted with the same colour.

Reviewer: ```suggestion
*Note*: UML notation for this Components View diagram is extended: both component and its interfaces are highlighted with the same color.
```

### Utility components:

#### Life Cycle
  - *Responsibility:*
    - Functional components manipulation
      -   creation
      -   destruction
      -   initialization
      -   start, stop
      -   binding
    - System and Utils-specifics initialization
    - *Relations*
    - Composes all available components
  - *Interfaces*
    -   Does not provide any external interfaces
  - *Behavior*

>     - ***Life Cycle*** creates all available in system components according configuration, binds components to components and starts each component internal routines.

Reviewer: ```suggestion
    - ***Life Cycle*** creates all available in system components according to configuration, binds components to components, and starts each component's internal routines.
```

    - ***HMIMessageHandler*** interface for accumulating ***HMIMessageObserver, HMIMessageSender*** and ***HMIMessageAdapter***
  - *Behavior*
    - Transferring RPC Messages between business-layer and configured transport.
  - *Constraints*
    - Processes messages from a single instance of HMI only.
    - HMI-transport need to be statically configurable with build flags.

### Application layer components:

#### Application Manager
  - *Responsibility*
    - Storing and providing mobile-related information
    - Mobile application state manipulation
  - *Relations*
    - Uses ***Commands***
    - Uses ***RPC Service***
    - Uses ***MediaManager***
    - Requires ***PolicyHandler** and **PolicyHandlerObserver (Policy)***
    - Requires ***ConnectionHandler** and **ConnectionHandlerObserver (Connection Handler)***
    - Requires ***SessionObserver (Connection Handler)***
    - Requires ***SecurityManagerListener (Security Manager*** component***)***
  - *Interfaces*
    - Provides ***ApplicationManager*** interface
  - *Behavior*
    - The component implements business logic of the SDL.
  - *Constraints*
    - *N/A*

#### RPC Service

>   - *Responcibility*

Reviewer: ```suggestion
  - *Responsibility*
```

    - Send Mobile/HMI RPCs
    - Handle Mobile/HMI RPCs
  - *Relations*
    - Uses ***PluginManager***
    - Uses ***RequestController***
    - Requires ***ProtocolHandler** and **ProtocolObserver (Protocol Handler)***
    - Requires ***HMIMessageObserver** and **HMIMessageSender (HMI Message Handler)***
  - *Interfaces*
    - Provides ***RPCService*** interface
  - *Behavior*

>     - The component implements RPC transfering functionality

Reviewer: ```suggestion
    - The component implements RPC transferring functionality
```

    - Stores information about plugin capabilities
    - Checks plugins capability to handle RPCs
  - *Relations*
    - Composed by ***Application Manager***
    - Composes ***Plugin***
  - *Interfaces*
    - Provides ***Plugin Manager*** interface
  - *Behavior*
    - Loads and manages plugins from specific directory.
  - *Constraints*
    - Able to load only RPC layer plugins

#### Plugin
  - *Responsibility*
    - Process Mobile/HMI RPC
  - *Relations*
    - Composed by ***Plugin Manager***
    - Uses ***Commands***
  - *Interfaces*
    - Provides ***Plugin*** interface
  - *Behavior*

>     - Handle specific RPC and process it's internal logic

Reviewer: ```suggestion
    - Handle specific RPC and process its internal logic
```
